src/NzbDrone.Api/Music/ArtistBulkImportModule.cs

```diff
@@ -0,0 +1,147 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using Nancy;
+using NzbDrone.Api.REST;
+using NzbDrone.Api.Extensions;
+using NzbDrone.Core.MediaCover;
+using NzbDrone.Core.MetadataSource;
+using NzbDrone.Core.Parser;
+using System.Linq;
+using System;
+using Marr.Data;
+using NzbDrone.Common.Extensions;
+using NzbDrone.Core.Datastore;
+using NzbDrone.Core.MediaFiles;
+using NzbDrone.Core.MediaFiles.EpisodeImport;
+using NzbDrone.Core.RootFolders;
+using NzbDrone.Common.Cache;
+using NzbDrone.Core.Music;
+
+namespace NzbDrone.Api.Music
+{
+
+    public class UnmappedComparer : IComparer<UnmappedFolder>
+    {
+        public int Compare(UnmappedFolder a, UnmappedFolder b)
+        {
+            return a.Name.CompareTo(b.Name);
+        }
+    }
+
+    public class MusicBulkImportModule : NzbDroneRestModule<ArtistResource>
+    {
+        private readonly ISearchForNewSeries _searchProxy;
+        private readonly IRootFolderService _rootFolderService;
+        private readonly IMakeImportDecision _importDecisionMaker;
+        private readonly IDiskScanService _diskScanService;
+		private readonly ICached<Core.Music.Artist> _mappedArtists;
+        private readonly IArtistService _artistService;
+
+        public MusicBulkImportModule(ISearchForNewSeries searchProxy, IRootFolderService rootFolderService, IMakeImportDecision importDecisionMaker,
+		                             IDiskScanService diskScanService, ICacheManager cacheManager, IArtistService artistService)
+            : base("/artist/bulkimport")
+        {
+            _searchProxy = searchProxy;
+            _rootFolderService = rootFolderService;
+            _importDecisionMaker = importDecisionMaker;
+            _diskScanService = diskScanService;
+			_mappedArtists = cacheManager.GetCache<Artist>(GetType(), "mappedArtistsCache");
+            _artistService = artistService;
+            Get["/"] = x => Search();
+        }
+
+
+        private Response Search()
+        {
+            if (Request.Query.Id == 0)
+            {
+                throw new BadRequestException("Invalid Query");
+            }
+
+            RootFolder rootFolder = _rootFolderService.Get(Request.Query.Id);
+
+            var unmapped = rootFolder.UnmappedFolders.OrderBy(f => f.Name).ToList();
+
+            var paged = unmapped;
+
+            var mapped = paged.Select(page =>
```

Comment:
Not sure of the C# style on this, but is this something that should be moved to a private function instead of anonymous function?